crates/runtime/src/starknet/state.rs

```diff
@@ -57,11 +47,6 @@ impl StateReader for DictStateReader {
     }
 
     fn get_compiled_class_hash(&self, class_hash: ClassHash) -> StateResult<CompiledClassHash> {
-        let compiled_class_hash = self
-            .class_hash_to_compiled_class_hash
-            .get(&class_hash)
-            .copied()
-            .unwrap_or_default();
-        Ok(compiled_class_hash)
+        Err(StateError::UndeclaredClassHash(class_hash))
```

Comment:
I'd prefer we throw an error like above here so it's immediately obvious where it's coming from. It could simplify some debugging in the future.